h/authorization.py

```diff
@@ -66,6 +82,48 @@ def is_web_client(request, client_id):
     return client_id == request.registry.web_client.client_id
 
 
+def normalize_authorization_token(request, token):  # bw compat
+    """Convert an Annotator Auth token into a standard JWT Bearer token."""
+    if 'issuedAt' in token:
+        token['iat'] = posix_seconds(iso8601.parse_date(token['issuedAt']))
+        if 'ttl' in token:
+            token['exp'] = token['iat'] + int(token['ttl'])
+            del token['ttl']
+        del token['issuedAt']
+
+    if 'consumerKey' in token:
+        token['iss'] = token['consumerKey']
+        del token['consumerKey']
+
+        if is_web_client(request, token['iss']):
+            token['aud'] = request.host_url
+
+    if 'userId' in token:
+        token['sub'] = token['userId']
+        del token['userId']
+
+
+def token_generator(request):  # bw compat
```

Comment:
This is *the* token generator. With this PR we write old tokens and read both. We either need to be able to disambiguate here between which client we're doing this for or we need to do the migration client-side by having a generation that supports the new format before we roll it out on the backend, wholesale replace this function here, and then remove the deprecated front end code that deals with the old format.